### DISCUSSION TITLE
How to build an extension in another language? eg: Rust

### DISCUSSION THREAD

**Discussion by theelderbeever** [duckdb/duckdb#8068] [Q&A] [answered]
I would like to be able to right some custom aggregation functions and don't really know C++ but I do know Rust. Is there any documentation on how to build extensions in other languages? Or how to build one at all?

I found other discussions that referenced these repos...
- https://github.com/duckdb/duckdb/tree/master/test/extension
- https://github.com/pdet/Scrooge-McDuck

**Comment by Mause** [comment, marked as answer]
The duckdb-rs project has a sample here: https://github.com/wangfenjin/duckdb-rs/blob/main/examples/hello-ext/main.rs
